icu4c/source/i18n/basictz.cpp

```diff
@@ -442,28 +449,25 @@ BasicTimeZone::getTimeZoneRulesAfter(UDate start, InitialTimeZoneRule*& initial,
                             break;
                         }
                     }
+                    if (U_FAILURE(status)) {
+                        return;
+                    }
                     int32_t asize = startTimes - idx;
                     if (asize > 0) {
-                        newTimes = (UDate*)uprv_malloc(sizeof(UDate) * asize);
-                        if (newTimes == NULL) {
+                        LocalMemory<UDate> newTimes((UDate*)uprv_malloc(sizeof(UDate) * asize));
```

Comment:
nit: The other `LocalMemory` objects use a `static_cast` with the malloc
```suggestion
                        LocalMemory<UDate> newTimes(static_cast<UDate*>(uprv_malloc(sizeof(UDate) * asize)));
```

Reply:
FYI
I don't _mind_ static_cast, but I don't think it adds much value, and tend to use the shorter C-style casts.

Different from const_cast and dynamic_cast which do add clear value.